src/ui/mainwindow.cc

```diff
@@ -1122,10 +1132,10 @@ MainWindow::~MainWindow()
 #endif
 }
 
-void MainWindow::addGlobalAction( QAction * action, const char * slot )
+void MainWindow::addGlobalAction( QAction * action, const std::function< void() > & slotFunc )
 {
   action->setShortcutContext( Qt::WidgetWithChildrenShortcut );
-  connect( action, SIGNAL( triggered() ), this, slot );
+  connect( action, &QAction::triggered, slotFunc );
```

Comment:
the third parameter  `this`?

Reply:
The `this` is part of the lambda, so it is not needed:

![image](https://github.com/xiaoyifang/goldendict-ng/assets/20123683/6c99f3c9-cd35-47e2-93e3-ac99b4513bbb)

This one only need 3 paramaters:

https://doc.qt.io/qt-6/qobject.html#connect-4

![image](https://github.com/xiaoyifang/goldendict-ng/assets/20123683/2e2c09e4-1593-4985-9380-538cf4ad7f73)

Reply:
in rare cases, the this has been deleted by outside logic .  the 3 parameter version will cause crash.

add `receiver` (in this case `this`) ,can avoid this crash. as it will not receive the signal anymore.

Reply:
Added.